File: backend/src/main/java/com/yagubogu/talk/dto/TalkResponse.java
package com.yagubogu.talk.dto;

import com.yagubogu.talk.domain.Talk;
import java.time.LocalDateTime;

public record TalkResponse(
        long id,
        Long memberId,
        String nickname,
        String favorite,
        String imageUrl,
        String content,
        LocalDateTime createdAt,
        boolean isMine
) {

    public static TalkResponse from(Talk talk, long memberId) {
        if (talk.getMember() == null) {
            return new TalkResponse(
                    talk.getId(),
                    null,
                    null,
                    null,
                    null,
                    talk.getContent(),
                    talk.getCreatedAt(),
                    false
            );
        }

        return new TalkResponse(
                talk.getId(),
                talk.getMember().getId(),
                talk.getMember().getNickname().getValue(),
                talk.getMember().getTeam().getShortName(),
                talk.getMember().getImageUrl(),
                talk.getContent(),
                talk.getCreatedAt(),
                talk.getMember().isSameId(memberId)
        );
    }

    public static TalkResponse hiddenFrom(TalkResponse talkResponse) {
        return new TalkResponse(
                talkResponse.id(),
                talkResponse.memberId(),
                talkResponse.nickname(),
                talkResponse.favorite(),
                talkResponse.imageUrl(),
                "숨김처리되었습니다",
                talkResponse.createdAt(),
                talkResponse.isMine()
        );
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/talk/TalkResponse.kt
package com.yagubogu.data.dto.response.talk

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TalkResponse(
    @SerialName("id")
    val id: Long, // 톡 메시지 ID
    @SerialName("memberId")
    val memberId: Long, // 메시지를 작성한 멤버의 ID
    @SerialName("nickname")
    val nickname: String, // 작성자의 닉네임
    @SerialName("favorite")
    val favorite: String, // 작성자의 응원팀 이름
    @SerialName("imageUrl")
    val imageUrl: String, // 작성자의 프로필 이미지 URL
    @SerialName("content")
    val content: String, // 채팅 메시지 본문
    @SerialName("createdAt")
    val createdAt: String, // 메시지 작성 시간 (ISO 8601 형식) 예: "2025-07-30T12:00:00"
    @SerialName("isMine")
    val isMine: Boolean, // 내가 작성한 메시지인지 여부
)
